routers/repo/issue.go

```diff
@@ -365,6 +365,16 @@ func RetrieveRepoMilestonesAndAssignees(ctx *context.Context, repo *models.Repos
 	}
 }
 
+// RetrieveRepoReviewers find all reviewers of a repository
+func RetrieveRepoReviewers(ctx *context.Context, repo *models.Repository, issuePosterID int64) {
+	var err error
+	ctx.Data["Reviewers"], err = repo.GetReviewers(ctx.User.ID, issuePosterID)
```

Comment:
```suggestion
	ctx.Data["Reviewers"], err = repo.GetReviewers(ctx.UserID, issuePosterID)
```
`ctx.UserID` always has a value, while `ctx.User` might not.

Reply:
``UserID`` is not in ctx type?
https://github.com/go-gitea/gitea/blob/2c25e75dca62850ddc328ed46c7b7d27b61be90b/modules/context/context.go#L32-L50

Reply:
I believe it's part of `*macaron.Context`, from which `context.Context` is derived from.

Reply:
？
https://github.com/go-gitea/gitea/blob/fbb0880d3760243fd56985c5e822facd68f58ebe/vendor/gitea.com/macaron/macaron/context.go#L82-L99

Reply:
I'm sorry, I must be confused with something else. I swear I remember several reviews in PRs like 'change ctx.User.ID into ctx.UserID because ctx.User is not always populated'.

In any case, could you please try and see what that code does when the user is not logged in? (e.g. in an all-public repo). Maybe it can never reach there?

Reply:
Maybe thinking of things like `issue.RepoID` vs `issue.Repo.ID`

Reply:
ok, I see, I will check them, Thanks

Reply:
@jolheiser Maybe it's that, mixed in with functions like `repo.MustOwner()`. Sorry about the extra work.

Reply:
@guillep2k  Hello, I think it's not a problem to use ``user.ID`` here, because it has been protected by a permission check in issue.go on Line 819, If have't login, these code will not be used, Do you think It's right？Thanks.
https://github.com/a1012112796/gitea/blob/dd900688fd3dec1f1100cf9260d9abc7afac8a28/routers/repo/issue.go#L818-L828